### ISSUE TITLE
[SCRIPT] Generate all channel schedules

### ISSUE THREAD

**Issue by irodimus** [justinemter/pseudo-channel#51]
Here's the script I pieced together to update all the channels. The script loops through the channels and updates all of them unless it finds the `running.pid` file in the channel directory, in which case it will skip it and move to the next. the channel that is skipped will update according to the value in the `pseudo_config.py` file so it helps to set this script to run as a cron job at that same time.

```
#!/bin/bash

# file: generate-channels-daily-schedules.sh

#----
# Simple script to generate the daily schedule for each individual channel. 
# 
#----

#---- 
# To Use:
# This script needs to be setup with a crontab entry that runs everyday at midnight or whatever interval you set to run.
#
# Configure something (a tv remote or alexa) to trigger this script. Make sure you move this script just 
# outside of the pseudo-channel directories:
# -------------------
# -channels/
# --pseudo-channel_1/
# ---startstop.sh
# --pseudo-channel_2/
# ---startstop.sh
# --pseudo-channel_3/
# ---startstop.sh
# --generate-channels-daily-schedules <--- on the same level as the 3 channels. 
#----

# Make sure that each channel dir ends with a "_" + an incrementing number as seen above.

#----BEGIN EDITABLE VARS----

pid_file=running.pid

python_to_use="$(which python)"

log_file=pseudo-channel.log

SCRIPT_PATH=$(dirname `which $0`)

CHANNEL_DIR_INCREMENT_SYMBOL="_"

#----END EDITABLE VARS-------

# Scan the dir to see how many channels there are, store them in an arr.
CHANNEL_DIR_ARR=( $(find . -maxdepth 1 -type d -name '*'"$CHANNEL_DIR_INCREMENT_SYMBOL"'*' -printf "%P\n") )

# If this script see's there are multiple channels, 
# then it loops through each channel. It then checks for running.pid
# in each channel and run the updates if not found for that channel.
if [ "${#CHANNEL_DIR_ARR[@]}" -gt 1 ]; then

	echo "+++++ There are ${#CHANNEL_DIR_ARR[@]} channels detected."

	for channel in "${CHANNEL_DIR_ARR[@]}"
	do
		
		if [ ! -e $channel/$pid_file ]; then

			cd "$channel"

			echo "+++++ Updating $channel"

			$python_to_use PseudoChannel.py -g >> $SCRIPT_PATH/$log_file

			cd ..

		else
	
			echo "+++++ Updating $channel... skipping, currently playing"

		fi

		sleep 1

	done

	echo "+++++ Daily channel schedules updated!"
	
fi

exit 0
```

**Comment by justinemter** [comment]
Ok cool, I just re-purposed my old `updatechannels.sh` to do the job but realize I am not checking for running pid's! I'll incorporate your logic, or if yours works out of the box as you say I'll just use that!

**Comment by irodimus** [comment]
I figured that already having PseudoChannel.py running, it would already update that channel anyway.

**Comment by irodimus** [comment]
Oh, and feel free to add, modify, remove, dissect, etc any part of this. I just couldn't wait and thought I'd share it for those who were looking for something :)

**Comment by justinemter** [comment]
Sounds great! I will absolutely give credit where credits due. Without testing it, it reads great. I may just add your script, put your username up top and add it to the scripts dir. Can you confirm it works with crontab? If so, does the user need to make crontab do anything special... like:

```bash
0 0 * * * cd /home/pi/channels/ bash ./generate-channels-daily-schedule.sh > /dev/null 2>&1
```

...I know often times cron needs to change dirs to make it work right (usually for py scripts). Maybe that's unnecessary?

**Comment by irodimus** [comment]
Honestly, as a macOS user, there are some small changes I made to the shell scripts for my use case to resolve issues that would occur when I attempted to run them in my setup. I use launchd to set the calendar interval on when to run these scripts and not necessarily crontab so I can't say for sure since PseudoChannel is run on my Mac.

It would probably be better all around to adjust these scripts to run on your setup so others who download the pseudo-channel project don't have to make changes like I did. I tried my best to use put those variables and commands back to the original as possible as seen in your other shell scripts.

Having said that, they should still work fine without any major tweaking.
